### ISSUE TITLE
glm and freeglut should be their own packages

### ISSUE THREAD

**Issue by GavinNL** [lasote/conan-sdl2#3]
Currently this script attempts to install glm, freeglut, and glew via apt-get, this will not work if you are using a non-ubuntu base system.

These three libraries should be their own conan packages, especially glm.

**Comment by lasote** [comment]
Totally agree, the sld2 support is VERY preliminary and requires too much time. If you could help would be great.

**Comment by GavinNL** [comment]
I was trying to make a package for glm, but I was unsure how to set up the py script for a header only library...I'll keep experimenting.

**Comment by lasote** [comment]
With the new "conan new" command you can setup a templated conanfile.py and allows a header only option:

```
conan new glm/1.0@gavinnl/testing --header --test
```

**Comment by a-teammate** [comment]
We also have issues with this in our CI
(But mostly because we do not want to give Jenkins sudo)
For us there is an easy workaround: bring the has_gl_installed_linux to work by installing the packages beforehand.

And although in the maintime 2 of these 3 packages are covered by conan, I do not think those dependencies are needed at all by SDL2.
I interpret these lines in question
```python
        if not self.has_gl_installed():
            if self.settings.os == "Linux":
                self.output.warn("GL is not installed in this machine! Conan will try to install it.")
                self.run("sudo apt-get update && sudo apt-get install -y freeglut3 freeglut3-dev libglew1.5-dev libglm-dev")
``` 
that we're actually looking for the package `libgl1-mesa-dev`.

Maybe giving outputting an error message directly (without trying to install mesa before) is the appropriate way to handle this.
One "could" [build mesa from source](https://www.gamingonlinux.com/wiki/Building_Mesa_from_source) in a conan package, but I dont know in how far that is what one wants on linux.
Maybe the current way is not the worst.

**TL;DR:** the title of this issue is missing the problem (the problem is to install opengl on linux).

**Comment by a-teammate** [comment]
Back on this:
This issue arises **only** because of some optional check `has_gl_installed()` in the conan recipe.
GL should always be installed (see https://github.com/conan-io/wishlist/issues/21), though.

In the case it isn't the errors will be much clearer than the errors when abusing sudo and apt to get an overly complicated test-case working (the one error will be probably be "could not find GL/gl.h specified in SDL2/SDL.h", when compiling the app).
the other error will appear when not using conan with sudo, or not on a debian derivate. or just on a machine missing apt. and it will appear in a strange place.
The opengl lib is furthermore already added to the cpp_libs array :) so no need to tune anything there :+1: 

I think this issue should be closed in the following manner:
- [ ] remove the code checking for has_gl_installed()
   * this test is actually doing sth else: its testing for glut, not gl.
   * afterwards the errors will be clearer for the dev
- [ ] remove the code which manually installs "GL"
